core/src/main/java/org/ldaptive/transport/DefaultSearchOperationHandle.java

```diff
@@ -30,6 +33,22 @@ public class DefaultSearchOperationHandle
   extends DefaultOperationHandle<SearchRequest, SearchResponse> implements SearchOperationHandle
 {
 
+  /** Predicate that requires any message except unsolicited. */
+  protected static final Predicate<Message> SEARCH_RESPONSE_TIMEOUT_CONDITION =
```

Comment:
It's not clear why `DEFAULT_RESPONSE_TIMEOUT_CONDITION` is insufficient for this case.

Reply:
The purpose of this patch is to treat search results differently from other operations. We want the `responseTimeout` to restart when other messages are received, not just the `Result` type. Would this be clearer if I narrowed the type checking?

Reply:
The question arose from ignorance of how LDAP responses are mapped to ldaptive API types. No changes needed, thanks for clarifying.